core/src/main/java/io/github/tcdl/Consumer.java

```diff
@@ -33,6 +35,9 @@
     private Clock clock;
 
     private MessageHandler messageHandler;
+
+    private Map<String, MessageHandler> messageHandlersByCorrelationId;
```

Comment:
Yes this map doesn't make sense for ResponderServer,but for ResponderServer messageHandler will be used.
I don't know what do you mean by this expression "lose encapsulation",but from my point of view this proposal have the following pros:
1) We have shorter chain of method calls Collector->Consumer->Adapter instead of Collector->Consumer->CollectorManager->Adapter
2)We don't have not understandable relationship between Collector,Consumer and CollectorManager.
3)We have less numbers of lines in our code.
4)And the most important for me is that in this case we have more clear and understandable code,from my point of view.